### PR TITLE
CDAP-1963 & CDAP-1964: Cube aggregation to have name & query in specific aggregation

### PR DESCRIPTION
**Pull request by abaranau**

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java [hunk 1]
```diff
@@ -140,19 +140,30 @@ SELECT count('read.ops')                                     << measure name and
...
-    Aggregation agg = findAggregation(query);
-    if (agg == null) {
-      throw new IllegalArgumentException("There's no data aggregated for specified tags to satisfy the query: " +
-                                           query.toString());
+    // 1) find aggregation to query
+    Aggregation agg;
+    if (query.getAggregation() != null) {
+      agg = aggregations.get(query.getAggregation());
+      if (agg == null) {
+        throw new IllegalArgumentException("Specified aggregation does not exist in cube: " + query.getAggregation());
```
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java]: should we list the available aggregation names here ? , so its useful for the user if there is a typo.

**Comment by gsps1** [comment]
not needed right now, but should we allow query by aggregation in MetricsHandler ?

**Comment by abaranau** [comment]
Probably not in metrics handler. But MetricStore should use it I think... We should eventually remove auto-choosing of aggregation

**Comment by gsps1** [comment]
just one minor comment, LGTM :+1:

**Comment by abaranau** [comment]
waiting for build: http://builds.cask.co/browse/CDAP-DUT1494-3

**Pull request merged**